addons/mail/static/src/new/core_ui/message.js

```diff
@@ -173,6 +173,7 @@ export class Message extends Component {
 
     get canReplyTo() {
         return (
+            this.message.originThread?.allowReplies &&
             this.props.messageToReplyTo &&
             (this.message.needaction || this.message.resModel === "mail.channel")
```

Comment:
Condition on `resModel === "mail.channel"` is actually a disguised condition on origin thread too (could be moved there instead)